lib/features/tag/tag.widget.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:flutter/widgets.dart';
+
+class CTag extends StatelessWidget {
+  CTag({
+    super.key,
+    required String label,
+    bool disabled = false,
+    bool filter = false,
+    CTagColors color = CTagColors.red,
+    CTagSize size = CTagSize.md,
+    this.onFilterPress,
+  }) : props = CTagProps(
+          disabled: disabled,
+          label: label,
+          filter: filter,
+          color: color,
+          size: size,
+        );
+
+  CTag.filter({
+    super.key,
+    required String label,
+    this.onFilterPress,
+    bool disabled = false,
+    CTagColors color = CTagColors.red,
+    CTagSize size = CTagSize.md,
+  }) : props = CTagProps(
+          disabled: disabled,
+          label: label,
+          filter: true,
+          color: color,
+          size: size,
+        );
+
+  /// {@macro tag_props}
+  final CTagProps props;
+
+  /// the callback when the filter icon is pressed
+  final VoidCallback? onFilterPress;
+
+  @override
+  Widget build(BuildContext context) {
+    return IntrinsicWidth(
+      child: AnimatedContainer(
+        duration: Duration(milliseconds: 200),
+        clipBehavior: Clip.antiAlias,
+        height: props.size == CTagSize.sm ? 20 : 28,
+        alignment: Alignment.center,
+        decoration: ShapeDecoration(
+          color: props.color.backgroundColor,
+          shape: const StadiumBorder(),
+        ),
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Padding(
+              padding: props.filter
+                  ? EdgeInsets.only(left: 8, right: 2)
+                  : EdgeInsets.symmetric(horizontal: 8),
+              child: Text(
+                props.label,
+                textAlign: TextAlign.center,
+                style: TextStyle(
+                  color: props.color.textColor,
+                  fontSize: 14,
+                  fontWeight: FontWeight.normal,
+                ),
+              ),
+            ),
+            props.filter
+                ? _CTagFilter(
+                    color: props.color,
+                    disabled: props.disabled,
+                    onPress: onFilterPress ?? () {},
+                    size: props.size,
+                  )
+                : const SizedBox.shrink(),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _CTagFilter extends StatefulWidget {
+  const _CTagFilter({
+    required this.disabled,
+    required this.color,
+    required this.onPress,
+    required this.size,
+  });
+
+  final bool disabled;
+  final CTagColors color;
+  final VoidCallback onPress;
+  final CTagSize size;
+
+  @override
+  State<_CTagFilter> createState() => _CTagFilterState();
+}
+
+class _CTagFilterState extends State<_CTagFilter> {
+  bool isHovered = false;
+  bool isPressed = false;
+
+  Color containerColor() {
+    if (widget.disabled) return CColors.gray20;
+
+    return isPressed //
```

Comment:
As I mentioned previously, separating styles logic from views will eliminate the ternary operators as much as possible when it comes to handling dynamic styling values

Another thing is that the current state of the project handles mobiles styles only (but the styling pattern used for styling enables us to scale and desktop styles easily. so for example i handle the focused event instead of mouse events using gesture detectors like this:
```
onTap: widget.props.onTap,
onTapDown: (_) => setState(() => focused = true),
onTapUp: (_) => setState(() => focused = false),
onTapCancel: () => setState(() => focused = false),
```

Soo, based on the previous two points, i believe it would be better if u handle the code using this approach (just like the other components):
```
  CWidgetState state = CWidgetState.enabled;

  bool focused = false;
  
    void _setStateVariables() {
    if (!isEnabled) {
      state = CWidgetState.disabled;
    } else if (_isEnabled && focused) {
      state = CWidgetState.focused;
    } else {
      state = CWidgetState.enabled;
    }
  }
```